composer/datasets/streaming/writer.py

```diff
@@ -4,19 +4,39 @@
 """:class:`StreamingDatasetWriter` is used to convert a list of samples into binary `.mds` files that can be read as a :class:`StreamingDataset`.
 """
 
+import gzip as gz
 import os
+from io import BufferedWriter
 from types import TracebackType
-from typing import Dict, Iterable, List, Optional, Type
+from typing import Any, Dict, Iterable, List, Optional, Type, Union
 
 import numpy as np
 from tqdm import tqdm
 
-from composer.datasets.streaming.format import (StreamingDatasetIndex, get_index_basename, get_shard_basename,
-                                                sample_dict_to_bytes)
+from composer.datasets.streaming.format import (StreamingDatasetIndex, get_compression_scheme_basename,
+                                                get_index_basename, get_shard_basename, sample_dict_to_bytes)
 
 __all__ = ['StreamingDatasetWriter']
 
 
+def _parse_compression_args(compression: Optional[str]) -> Dict[str, Any]:
```

Comment:
Would prefer to return a tuple of (compression_scheme, compresslevel) rather than a dict -- since this is an internal API, we can change it later as we support additional compression options.

```suggestion
def _parse_compression_args(compression: Optional[str]) -> Tuple[str, int]:
```